torch/testing/_internal/common_methods_invocations.py

```diff
@@ -529,6 +529,55 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
                             dim=(0, 1))))
         return inputs
 
+def sample_inputs_add(op_info, device, dtype, requires_grad, **kwargs):
+    for_inplace_variant = kwargs.get('for_inplace_variant', False)
+    tests_list = [
+        ((S, S, S), (S, S, S)),
+        ((S, S, S), (S, S)),
+        ((), ()),
+        ((S, S, S), ()),
+    ]
+    tests_require_resizing = [
+        ((S, S), (S, S, S)),
+        ((), (S, S, S)),
+        ((S, 1, S), (M, S)),
+    ]
+    test_cases = [*tests_list] if for_inplace_variant else [*tests_list, *tests_require_resizing]
+    inputs = list(SampleInput(make_tensor(first_shape, device=device, dtype=dtype,
+                                          requires_grad=requires_grad),
+                              args=(make_tensor(second_shape, device=device, dtype=dtype,
+                                    requires_grad=requires_grad),))
+                  for first_shape, second_shape in test_cases)
+    if dtype in [torch.bfloat16, torch.float16, torch.float32, torch.float64]:
+        args_list = [
```

Comment:
Actually instead of checking for whether the tensor is floating point or complex, just create the type appropriate scalar value upfront (3, 3.14, 3.14 - 3.14j) and then add these two cases to the previous lists (one to each list):

```
tests_list = (
  ...
  ((S, S, S), scalar),
  ((), scalar)
)
```

then enumerate the test cases:

```
samples = []
for shape, shape_or_scalar in test_cases:
  arg = shape_or_scalar
  if isinstance(shape_or_scalar, tuple):
   arg = make_tensor(shape_or_scalar, device=device, dtype=dtype, requires_grad=requires_grad)
  
  inp = make_tensor(shape, device=device, dtype=dtype, requires_grad=requires_grad)
  sample = SampleInput(inp, args=(arg,))
  samples.append(sample)

return tuple(samples)
```